crates/typst-syntax/src/parser.rs

```diff
@@ -1138,18 +1138,40 @@ fn module_import(p: &mut Parser) {
     let m = p.marker();
     p.assert(SyntaxKind::Import);
     code_expr(p);
-    if p.eat_if(SyntaxKind::Colon) && !p.eat_if(SyntaxKind::Star) {
-        import_items(p);
+    if p.eat_if(SyntaxKind::As) {
+        // allow renaming a full module import
+        p.expect(SyntaxKind::Ident);
+        p.wrap(m, SyntaxKind::ModuleImport);
+
+        // friendlier error if the user tries to rename and import items at the
+        // same time
+        if p.at(SyntaxKind::Colon) {
+            p.unexpected();
+            p.hint("cannot import the renamed module and its items at the same time");
+            p.hint("try importing the module's items in a separate `import` expression");
+        }
+    } else {
+        if p.eat_if(SyntaxKind::Colon) && !p.eat_if(SyntaxKind::Star) {
+            import_items(p);
+        }
+        p.wrap(m, SyntaxKind::ModuleImport);
     }
-    p.wrap(m, SyntaxKind::ModuleImport);
```

Comment:
I think it's fine to do the wrapping here (for both cases) and the use `else if` above. it doesn't matter if the colon (and potentially the import items if my suggestion from above is applied) are part of the import if it's malformed. in fact, one could say that it makes more sense.